In [79]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import re
%matplotlib inline

In [94]:
wellcome = pd.read_csv('../../__DATA__/WELLCOME/WELLCOME_APCspend2013_forThinkful.csv', encoding="latin-1")

In [95]:
wellcomeraw = pd.read_csv('../../__DATA__/WELLCOME/WELLCOME_APCspend2013_forThinkful.csv', encoding="latin-1")

In [69]:
wellcomeraw.head(5)

PMID/PMCID Publisher           Journal title  \
0                    NaN       CUP  PSYCHOLOGICAL MEDICINE   
1             PMC3679557       ACS       BIOMACROMOLECULES   
2  23043264  PMC3506128        ACS              J MED CHEM   
3    23438330 PMC3646402       ACS              J MED CHEM   
4   23438216 PMC3601604        ACS              J ORG CHEM   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  COST (£) charged to Wellcome (inc VAT when charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88

In [8]:
len(wellcome)

2127

In [105]:
wellcome = wellcome.rename(columns={'COST (£) charged to Wellcome (inc VAT when charged)': 'Cost'})

In [106]:
wellcome.head(5)

PMID/PMCID Publisher           Journal title  \
0                    NaN       CUP  PSYCHOLOGICAL MEDICINE   
1             PMC3679557       ACS       BIOMACROMOLECULES   
2  23043264  PMC3506128        ACS              J MED CHEM   
3    23438330 PMC3646402       ACS              J MED CHEM   
4   23438216 PMC3601604        ACS              J ORG CHEM   

                                       Article title     Cost  
0  Reduced parahippocampal cortical thickness in ...     0.00  
1  Structural characterization of a Model Gram-ne...  2381.04  
2  Fumaroylamino-4,5-epoxymorphinans and related ...   642.56  
3  Orvinols with mixed kappa/mu opioid receptor a...   669.64  
4  Regioselective opening of myo-inositol orthoes...   685.88

In [96]:
upperlist = []
for i in wellcome['Journal title']:
    upperlist.append(str(i).upper())

upperlist    
wellcome['Journal title'] = upperlist

wellcome.head(10)

PMID/PMCID                                     Publisher  \
0                    NaN                                           CUP   
1             PMC3679557                                           ACS   
2  23043264  PMC3506128                                            ACS   
3    23438330 PMC3646402                                           ACS   
4   23438216 PMC3601604                                            ACS   
5             PMC3579457                                           ACS   
6             PMC3709265                                           ACS   
7   23057412 PMC3495574                                            ACS   
8      PMCID: PMC3780468  ACS (Amercian Chemical Society) Publications   
9      PMCID: PMC3621575  ACS (Amercian Chemical Society) Publications   

                    Journal title  \
0          PSYCHOLOGICAL MEDICINE   
1               BIOMACROMOLECULES   
2                      J MED CHEM   
3                      J MED CHEM   
4                      J ORG CHEM   
5  JOURNAL OF MEDICINAL CHEMISTRY   
6    JOURNAL OF PROTEOME RESEARCH   
7                       MOL PHARM   
8            ACS CHEMICAL BIOLOGY   
9            ACS CHEMICAL BIOLOGY   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   
5  Comparative Structural and Functional Studies ...   
6  Mapping Proteolytic Processing in the Secretom...   
7  Quantitative silencing of EGFP reporter gene b...   
8  A Novel Allosteric Inhibitor of the Uridine Di...   
9  Chemical proteomic analysis reveals the drugab...   

  COST (£) charged to Wellcome (inc VAT when charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88   
5                                           £2392.20   
6                                           £2367.95   
7                                            £649.33   
8                                           £1294.59   
9                                           £1294.78

In [32]:
jtitles = sorted(wellcome['Journal title'].dropna().unique().tolist())
jtitles

['ACS Chemical Biology',
 'ACS Chemical Neuroscience',
 'ACS NANO',
 'ACS Nano',
 'ACTA F',
 'AGE',
 'AIDS',
 'AIDS Behav',
 'AIDS Care',
 'AIDS Journal',
 'AIDS Research and Therapy',
 'AIDS UK',
 'ASN Neuro',
 'Academy of Nutrition and Dietetics',
 'Acta Crystallographica Section D,  Biological Crystallography ',
 'Acta Crystallographica Section D: Biological Crystallography',
 'Acta Crystallographica Section F: Structural Biology and Crystallization Communications',
 'Acta Crystallographica, Section D',
 'Acta Crystallography D',
 'Acta D',
 'Acta Dermato Venereologica',
 'Acta Diabetologica',
 'Acta Neuropathol',
 'Acta Neuropathologica',
 'Acta Opthalmologica',
 'Acta Physiol',
 'Addiction',
 'Advances in Experimental Medicine and Biology',
 'Age',
 'Age and Ageing',
 'Ageing & Society',
 'Aging Cell',
 'Aids Care',
 'Alcohol and Alcoholism',
 'Alimentrary Pharmacology & Therapeutics',
 'Am J Bioeth',
 'Am J Trop Med Hyg',
 'Americal Journal of Psychiatry',
 'American Chemical Soc

In [68]:
wellcome.groupby('Journal title').count().sort_values(by='Article title', ascending=False)['Article title'].head(5)

Journal title
PLOS ONE                           190
JOURNAL OF BIOLOGICAL CHEMISTRY     53
NEUROIMAGE                          29
PLOS GENETICS                       24
PLOS PATHOGENS                      24
Name: Article title, dtype: int64

The top 5 journals are PLoS One, Journal of Biological Chemistry, Neuroimage, PloS Genetics, and PLoS Pathogens with 190, 53, 29, 24, and 24 articles respectively.

In [110]:
costlist = []
non_decimal = re.compile(r'[^\d.]+')
for i in wellcome['Cost']:
    costlist.append(float(non_decimal.sub('', i)))
wellcome['COST (£) charged to Wellcome (inc VAT when charged)'] = costlist

TypeError: expected string or bytes-like object

In [111]:
wellcome.groupby('Journal title').mean()

Cost
Journal title                                                    
ACADEMY OF NUTRITION AND DIETETICS                    2379.540000
ACS CHEMICAL BIOLOGY                                  1418.186000
ACS CHEMICAL NEUROSCIENCE                             1186.800000
ACS NANO                                               668.140000
ACTA CRYSTALLOGRAPHICA SECTION D,  BIOLOGICAL C...     771.420000
ACTA CRYSTALLOGRAPHICA SECTION D: BIOLOGICAL CR...     773.740000
ACTA CRYSTALLOGRAPHICA SECTION F: STRUCTURAL BI...     796.635000
ACTA CRYSTALLOGRAPHICA, SECTION D                      757.180000
ACTA CRYSTALLOGRAPHY D                                 774.190000
ACTA D                                                 750.160000
ACTA DERMATO VENEREOLOGICA                             653.960000
ACTA DIABETOLOGICA                                    2336.280000
ACTA F                                                 754.900000
ACTA NEUROPATHOL                                      1901.040000
ACTA NEUROPATHOLOGICA                                 2161.063333
ACTA OPTHALMOLOGICA                                   2270.160000
ACTA PHYSIOL                                          1991.500000
ADDICTION                                             2136.225000
ADVANCES IN EXPERIMENTAL MEDICINE AND BIOLOGY         1928.457500
AGE                                                   2170.140000
AGE AND AGEING                                        2040.000000
AGEING & SOCIETY                                      1695.000000
AGING CELL                                            1970.570000
AIDS                                                  2059.306667
AIDS BEHAV                                            1834.770000
AIDS CARE                                             2294.225000
AIDS JOURNAL                                          2015.720000
AIDS RESEARCH AND THERAPY                             1240.000000
AIDS UK                                               1836.920000
ALCOHOL AND ALCOHOLISM                                2040.000000
...                                                           ...
TRAFFIC                                               2129.338333
TRANSACTIONS OF THE ROYAL SOCIETY OF TROPICAL M...    1524.080000
TRANSLATIONAL PSYCHIATRY                              2880.000000
TRENDS IN COGNITIVE SCIENCE                           2340.630000
TRENDS IN GENETICS                                    2370.935000
TRENDS IN MICROBIOLOGY                                2291.330000
TRENDS IN MOLECULAR MEDICINE                          2248.330000
TRENDS IN NEUROSCIENCE                                2402.810000
TRENDS IN NEUROSCIENCES                             501212.980000
TRENDS IN PARASITOLOGY                              251692.990000
TRIALS                                              250908.975000
TRIALS                                                1165.000000
TROP MED INT HEALTH                                   1834.770000
TROPICAL ANIMAL HEALTH & PRODUCTION                   2054.780000
TROPICAL MEDICINE AND INTERNATIONAL HEALTH            1843.163333
TUBERCULOSIS                                          2161.255000
URBAN HISTORY                                         2034.000000
UROLITHIASIS                                          1112.400000
VACCINE                                               1623.394000
VASCULAR PHARMACOLOGY                                 2352.530000
VETERINARY MICROBIOLOGY                               2451.490000
VETERINARY PARASITOLOGY                             501243.585000
VETERINARY RECORD                                     2040.000000
VETERINARY RESEARCH                                      9.930000
VIROLOGY                                              1928.505000
VIROLOGY JOURNAL                                      1242.000000
VIRUS RESEARCH                                        1947.090000
VISION RESEARCH                                     500727.590000
VISUAL NEUROSCIENCE                

In [116]:
#wellcome.sort_values(by=['Cost'], ascending=False).head(100)

In [119]:
wellcome.groupby('Journal title').std().replace(float(0),np.nan).dropna()

Cost
Journal title                                                    
ACS CHEMICAL BIOLOGY                                   507.309560
ACS NANO                                                35.708892
ACTA CRYSTALLOGRAPHICA SECTION F: STRUCTURAL BI...      15.605847
ACTA NEUROPATHOLOGICA                                  244.811798
ADDICTION                                              306.481292
ADVANCES IN EXPERIMENTAL MEDICINE AND BIOLOGY            0.005000
AGE                                                    237.785868
AIDS                                                   281.067979
AIDS CARE                                              126.415851
AMERICAN ETHNOLOGIST                                     0.247487
AMERICAN JOURNAL OF EPIDEMIOLOGY                       196.299092
AMERICAN JOURNAL OF HUMAN GENETICS                     703.501205
AMERICAN JOURNAL OF TRANSPLANTATION                    278.022730
AMERICAN JOURNAL OF TROPICAL MEDICINE & HYGIENE          2.852110
ANALYTICAL CHEMISTRY                                   838.678951
ANIMAL BEHAVIOUR                                         7.071068
ANNALS OF NEUROLOGY                                    424.183550
ANNALS OF THE NEW YORK ACADEMY OF SCIENCES             560.926596
ANTIMICROBIAL AGENTS AND CHEMOTHERAPY                  115.343350
ANTIOXIDANTS AND REDOX SIGNALING                       598.707312
ARCHIVES OF DISEASE IN CHILDHOOD                       173.205081
ARTHRITIS & RHEUMATISM                                 484.592530
ASIAN JOURNAL OF PSYCHIATRY                            288.980399
ATHEROSCLEROSIS                                         61.044528
BEHAVIOR RESEARCH AND THERAPY                           87.172124
BEHAVIORAL NEUROSCIENCE                                109.112900
BIOCHEMICAL AND BIOPHYSICAL RESEARCH COMMUNICAT...     545.179328
BIOCHEMICAL JOURNAL                                    149.936855
BIOCHEMICAL SOCIETY TRANSACTIONS                       624.499800
BIOETHICS                                               40.372602
...                                                           ...
RESPIRATORY RESEARCH                                   205.270237
RNA                                                    139.667731
RNA BIOLOGY                                             85.387799
SCHIZOPHRENIA BULLETIN                                   5.303301
SCIENCE-RESEARCH                                       194.518004
SOCIAL COGNITIVE AND AFFECTIVE NEUROSCIENCE            138.564065
SOCIAL HISTORY OF MEDICINE                              30.000000
SOCIAL PSYCHIATRY AND PSYCHIATRIC EPIDEMIOLOGY         172.267124
SOCIAL SCIENCE AND MEDICINE                            600.217876
SOCIOLOGY OF HEALTH & ILLNESS                           59.453538
STEM CELLS                                             220.013917
STRUCTURE                                              166.835165
THE AMERICAN JOURNAL OF HUMAN GENETICS                 868.682815
THE EMBO JOURNAL                                      1103.086579
THE FASEB JOURNAL                                      406.966595
THE JOURNAL OF BIOLOGICAL CHEMISTRY                    138.729897
THE JOURNAL OF NEUROSCIENCE                            591.763523
THE LANCET                                             240.024915
THE LANCET NEUROLOGY                                  1018.233765
TRAFFIC                                                340.691688
TRENDS IN GENETICS                                       3.995153
TRENDS IN NEUROSCIENCES                             705389.954206
TRENDS IN PARASITOLOGY                              498870.709384
TRIALS                                              499393.359606
TROPICAL MEDICINE AND INTERNATIONAL HEALTH             161.141743
TUBERCULOSIS                                           228.133861
VACCINE                                                394.821486
VETERINARY PARASITOLOGY                             705346.672200
VIROLOGY                           

In [120]:
wellcome.groupby('Journal title').median()

Cost
Journal title                                                 
ACADEMY OF NUTRITION AND DIETETICS                    2379.540
ACS CHEMICAL BIOLOGY                                  1294.590
ACS CHEMICAL NEUROSCIENCE                             1186.800
ACS NANO                                               668.140
ACTA CRYSTALLOGRAPHICA SECTION D,  BIOLOGICAL C...     771.420
ACTA CRYSTALLOGRAPHICA SECTION D: BIOLOGICAL CR...     773.740
ACTA CRYSTALLOGRAPHICA SECTION F: STRUCTURAL BI...     796.635
ACTA CRYSTALLOGRAPHICA, SECTION D                      757.180
ACTA CRYSTALLOGRAPHY D                                 774.190
ACTA D                                                 750.160
ACTA DERMATO VENEREOLOGICA                             653.960
ACTA DIABETOLOGICA                                    2336.280
ACTA F                                                 754.900
ACTA NEUROPATHOL                                      1901.040
ACTA NEUROPATHOLOGICA                                 2250.970
ACTA OPTHALMOLOGICA                                   2270.160
ACTA PHYSIOL                                          1991.500
ADDICTION                                             2136.225
ADVANCES IN EXPERIMENTAL MEDICINE AND BIOLOGY         1928.460
AGE                                                   2170.140
AGE AND AGEING                                        2040.000
AGEING & SOCIETY                                      1695.000
AGING CELL                                            1970.570
AIDS                                                  1968.630
AIDS BEHAV                                            1834.770
AIDS CARE                                             2316.010
AIDS JOURNAL                                          2015.720
AIDS RESEARCH AND THERAPY                             1240.000
AIDS UK                                               1836.920
ALCOHOL AND ALCOHOLISM                                2040.000
...                                                        ...
TRAFFIC                                               2273.455
TRANSACTIONS OF THE ROYAL SOCIETY OF TROPICAL M...    1524.080
TRANSLATIONAL PSYCHIATRY                              2880.000
TRENDS IN COGNITIVE SCIENCE                           2340.630
TRENDS IN GENETICS                                    2370.935
TRENDS IN MICROBIOLOGY                                2291.330
TRENDS IN MOLECULAR MEDICINE                          2248.330
TRENDS IN NEUROSCIENCE                                2402.810
TRENDS IN NEUROSCIENCES                             501212.980
TRENDS IN PARASITOLOGY                                2384.110
TRIALS                                                1253.250
TRIALS                                                1165.000
TROP MED INT HEALTH                                   1834.770
TROPICAL ANIMAL HEALTH & PRODUCTION                   2054.780
TROPICAL MEDICINE AND INTERNATIONAL HEALTH            1883.625
TUBERCULOSIS                                          2161.255
URBAN HISTORY                                         2034.000
UROLITHIASIS                                          1112.400
VACCINE                                               1448.380
VASCULAR PHARMACOLOGY                                 2352.530
VETERINARY MICROBIOLOGY                               2451.490
VETERINARY PARASITOLOGY                             501243.585
VETERINARY RECORD                                     2040.000
VETERINARY RESEARCH                                      9.930
VIROLOGY                                              1928.505
VIROLOGY JOURNAL                                      1242.000
VIRUS RESEARCH                                        1947.090
VISION RESEARCH                                     500727.590
VISUAL NEUROSCIENCE                                   2034.000
ZOONOSES AND PUBLIC HEALTH                            2272.150

[929 rows x 1 columns]